network/src/sync/blocks/aggro.rs

```diff
@@ -0,0 +1,201 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{sync::Arc, time::Duration};
+
+use crate::{Cache, Node, Payload, Peer, SyncBase, SyncInbound};
+use anyhow::*;
+use indexmap::IndexSet;
+use snarkos_storage::Digest;
+use snarkvm_dpc::BlockHeaderHash;
+use tokio::{
+    sync::{mpsc, RwLock},
+    time::Instant,
+};
+
+pub struct SyncAggro {
```

Comment:
having seen all the `SyncX` objects I know what their names mean, but it would be nice to have short doc comments on them for future readers